core/src/HIP/Kokkos_HIP_KernelLaunch.hpp

```diff
@@ -76,28 +78,31 @@ void *hip_resize_scratch_space(std::int64_t bytes, bool force_shrink = false);
 
 template <typename DriverType>
 __global__ static void hip_parallel_launch_constant_memory() {
+// cannot use global constants in HCC
+#ifdef __HCC__
   __device__ __constant__ unsigned long kokkos_impl_hip_constant_memory_buffer
       [Kokkos::Experimental::Impl::HIPTraits::ConstantMemoryUsage /
        sizeof(unsigned long)];
+#endif
 
-  const DriverType &driver = *(reinterpret_cast<const DriverType *>(
+  const DriverType *driver = (reinterpret_cast<const DriverType *>(
```

Comment:
make this `const DriverType* const` not sure if it makes a different but might.

Reply:
1672f85b